lib/optimizely/custom_attribute_condition_evaluator.rb

```diff
@@ -73,13 +111,48 @@ def exact_evaluator(condition)
 
       user_provided_value = @user_attributes[condition['name']]
 
-      if user_provided_value.is_a?(Numeric) && condition_value.is_a?(Numeric)
-        return true if condition_value.to_f == user_provided_value.to_f
+      # Return true when 2 == 2.0
+      if condition_value.is_a?(Numeric) && user_provided_value.is_a?(Numeric) &&
+         Helpers::Validator.finite_number?(condition_value) &&
+         Helpers::Validator.finite_number?(user_provided_value)
+
+        return true if user_provided_value == condition_value
```

Comment:
Do we need to do some kind of special comparison instead of `user_provided_value == condition_value` in order to account for cases in which `user_provided_value` and `condition_value` are of different numeric types?

If not, I think it'd be cleaner to rely on the equality check that already occurs at the end of this function. At the very least, this would be consistent with most of our other SDKs. In exchange, we'd have to modify `Helpers::Validator.same_types?` to return true in the case of different numeric types.

Reply:
+1. It will be much cleaner that way.